### PR TITLE
Convert System.Xml.XDocument.Properties.Tests to use Xunit (Part 1)

### PR DESCRIPTION
**Pull request by Maxwe11**
Progress against #642

I recommend to review without whitespace changes
https://github.com/dotnet/corefx/pull/3854/files?w=1

/cc: @krwq

### CODE REVIEW

File: src/System.Xml.XDocument/tests/Properties/FunctionalTests.cs [hunk 1]
```diff
@@ -158,141 +70,6 @@ public override void AddChildren()
...
-                    this.AddChild(new TestVariation(BoolConvert) { Attribute = new VariationAttribute("bool - concat from children - 1") { Params = new object[] { "<A> <B><D><![CDATA[1]]></D></B><C> </C></A>", true }, Priority = 1 } });
-                    this.AddChild(new TestVariation(BoolConvert) { Attribute = new VariationAttribute("bool - concat from children - false") { Params = new object[] { "<A><B>fa<D><![CDATA[l]]></D></B><C>se</C></A>", false }, Priority = 1 } });
-                    this.AddChild(new TestVariation(BoolConvert) { Attribute = new VariationAttribute("bool - concat from children - 0") { Params = new object[] { "<A><B> <D><![CDATA[ ]]></D></B><C>0</C></A>", false }, Priority = 1 } });
-                    this.AddChild(new TestVariation(BoolConvertInvalid) { Attribute = new VariationAttribute("bool - Invalid - empty") { Params = new object[] { "<A></A>" }, Priority = 2 } });
-                    this.AddChild(new TestVariation(BoolConvertInvalid) { Attribute = new VariationAttribute("bool - Invalid - capital T") { Params = new object[] { "<A>True</A>" }, Priority = 2 } });
```
Comment (Maxwe11) [inline src/System.Xml.XDocument/tests/Properties/FunctionalTests.cs]: :exclamation:  this test case is invalid

File: src/System.Xml.XDocument/tests/Properties/FunctionalTests.cs [hunk 2]
```diff
@@ -158,141 +70,6 @@ public override void AddChildren()
...
-                    this.AddChild(new TestVariation(BoolQConvert) { Attribute = new VariationAttribute("bool? - concat from children - 0") { Params = new object[] { "<A><B> <D><![CDATA[ ]]></D></B><C>0</C></A>", false }, Priority = 1 } });
-                    this.AddChild(new TestVariation(BoolQConvert) { Attribute = new VariationAttribute("bool? - direct - true") { Params = new object[] { "<A>true</A>", true }, Priority = 0 } });
-                    this.AddChild(new TestVariation(BoolQConvert) { Attribute = new VariationAttribute("bool? - direct - 1") { Params = new object[] { "<A>1</A>", true }, Priority = 0 } });
-                    this.AddChild(new TestVariation(BoolQConvert) { Attribute = new VariationAttribute("bool? - direct - false") { Params = new object[] { "<A>false</A>", false }, Priority = 0 } });
-                    this.AddChild(new TestVariation(BoolQConvertInvalid) { Attribute = new VariationAttribute("bool? - Invalid - capital T") { Params = new object[] { "<A a='1'>True</A>" }, Priority = 2 } });
```
Comment (Maxwe11) [inline src/System.Xml.XDocument/tests/Properties/FunctionalTests.cs]: :exclamation: this test case is invalid

File: src/System.Xml.XDocument/tests/Properties/DeepEquals.cs [hunk 3]
```diff
@@ -1,407 +1,390 @@
...
+        [InlineData("PI", "", "PI", "", true)]                  // data = ''
+        [InlineData("PI", "click", "PI", "", false)]            // data1!=data2
+        [InlineData("AAAAP", "click", "AAAAQ", "click", false)] // target1!=target2
+        [InlineData("AAAAP", "AAAAQ", "AAAAP", "AAAAQ", true)]  // hashconflict I
+        [InlineData("PA", "PA", "PI", "PI", false)]             // data=target
+        [InlineData("AAAAP", "AAAAQ", "AAAAQ", "AAAAP", false)] // hashconflict II
+        [Theory]
+        public void ProcessingInstruction(string target1, string data1, string target2, string data2, bool checkHashCode)
         {
-            public partial class DeepEqualsTests : XLinqTestCase
```
Comment (stephentoub) [inline src/System.Xml.XDocument/tests/Properties/DeepEquals.cs]: There are a few large blocks of code being deleted here, e.g. DeepEqualsTests and ImplicitConversionsElem, and I'm not seeing what their replacements are.  I'm probably just missing it?

**Comment by stephentoub** [comment]
One question, but otherwise looks good.  Thanks for doing this.

File: src/System.Xml.XDocument/tests/Properties/DeepEquals.cs (hunk 3 above)
Comment (Maxwe11) [inline src/System.Xml.XDocument/tests/Properties/DeepEquals.cs]: >There are a few large blocks of code being deleted here

I recall only one [such place](https://github.com/dotnet/corefx/pull/3854/files?w=1#diff-013916273010d7a23c6415c3d2c05e77L47). These code not used anywhere.


>ImplicitConversionsElem

All code there has relevant replacements except a couple test cases
https://github.com/dotnet/corefx/pull/3854#discussion_r42038208
https://github.com/dotnet/corefx/pull/3854#discussion_r42038377

>DeepEqualsTests 

All code there has relevant replacements

Comment (stephentoub) [inline src/System.Xml.XDocument/tests/Properties/DeepEquals.cs]: I see, ok, thanks.

**Pull request merged**